### PR TITLE
Get transaction info in get-kafka-instance-state.sh

### PR DESCRIPTION
**Pull request by grdryn**

### CODE REVIEW

File: scripts/CEE/get-kafka-instance-state/metadata.yaml
Comment (k-wall) [inline scripts/CEE/get-kafka-instance-state/metadata.yaml:89]: is there a use-case for running the command against fewer than the number of brokers the instance has?  If no, I think the script should work out the number of brokers that exist, and run it against all.

File: scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh
Comment (k-wall) [inline scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh:48]: is this a dup with line 45?

File: scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh
Comment (k-wall) [inline scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh:52]: Does `${inspectDir}/kafka-transactions-describe.txt` filename need the transaction-id interpolated into it to make it unique.

File: scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh
Comment (SamBarker) [inline scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh:55]: Is there any reason to do this sequentially? It would be faster to run each  broker in the background.

File: scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh
Comment (SamBarker) [inline scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh:46]: Would it be worth gathering all transaction related files into a specific sub directory within `inspectDir`

File: scripts/CEE/get-kafka-instance-state/metadata.yaml
Comment (grdryn) [reply scripts/CEE/get-kafka-instance-state/metadata.yaml:89]: > is there a use-case for running the command against fewer than the number of brokers the instance has? If no, I think the script should work out the number of brokers that exist, and run it against all.

Good idea, thanks. I'll try that.

> Also I notice that the [--broker-id BROKER_ID] is marked as optional, do we actually need to pass it at all?

The `find-hanging` command requires either `--topic` or `--broker-id` to limit the scope of the search. It seems easier to pass the broker id.

File: scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh
Comment (grdryn) [reply scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh:51]: > I think your intent is to iterate across all the transaction id in the kafka-transactions-list.txt and run a describe on each, right? There's a problem in that. if the kafka instance has for whatever reason accumulated lots of transactions, get-managed-instance-state might take an excessive length of time to run. I suggest an extra parameter that limits the number of transaction describes that will be performed. Default this to something like 25.

Hmm, I hadn't thought of that as a possible problem. :thinking: I'm leaning more towards not running the `describe` on any transactions than just choosing a limited number for now. If we do need that information (I'm not sure that we currently do), then there's no way of knowing if the transaction(s) we actually do want it for would be returned as part of the limited set of 25 results, and we wouldn't really have a way to know what number we have to set to get the results we do want. Maybe we need a separate managed script that just wraps that `describe` sub-command. This is what I was thinking about as a possibility for the `abort` command also. WDYT?

> Separately, your scripting looks suspicious. kafka-transactions-list.txt is on the client, but you are passing in within the shell command executed within the pod. It won't be able to access the file.

Yes, I started testing this after pushing the WIP PR and noticed that too :)

File: scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh
Comment (grdryn) [reply scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh:52]: Either that, or each invocation should append to the file rather than overwriting it.

File: scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh
Comment (grdryn) [reply scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh:55]: I guess that could be done, we'd just need to await them before tar-streaming the directory into the logs.

File: scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh
Comment (grdryn) [reply scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh:46]: Maybe :thinking: I'll consider if that's worth it after I (maybe) move the `describe` stuff out of here to a separate script.

Comment (grdryn) [reply scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh]: Done, thanks for the suggestion!

**Comment by grdryn** [comment]
I've updated this to pull the describe part out into a separate script that can be called on a specific transaction, and addressed the remaining comments. @k-wall @SamBarker Please take another look if you have a moment, and let me know what you think of this approach. Thanks!

(I don't know what's going on with that failing build...I don't have access to see the logs :disappointed:)

**Comment by grdryn** [comment]
@SamBarker could you take a look at this when you have a moment, to see if you're happy with it now? Thanks!

**Review by SamBarker — APPROVED** [review]
LGTM

**Comment by grdryn** [comment]
@wanghaoran1988, @MitaliBhalla could you please take a look at this and approve if you have no objections? Thanks a lot! :slightly_smiling_face:

**Comment by grdryn** [comment]
/hold

I think I need to make another change here

**Comment by grdryn** [comment]
/hold cancel

I've updated this to be fine again. I just realised while doing some extra testing that I had specified `wait -f`, and in the version of Bash in the container image, `wait` doesn't yet have `-f`, so `wait` is enough.

File: scripts/CEE/describe-kafka-transaction/metadata.yaml
Comment (wanghaoran1988) [inline scripts/CEE/describe-kafka-transaction/metadata.yaml:17]: Do you need this permission at cluster scope?

Comment (grdryn) [reply scripts/CEE/describe-kafka-transaction/metadata.yaml:17]: @wanghaoran1988 I _think_ so, but I'm happy to hear of any suggestions that you have. My understanding is that the script will run in the openshift-managed-scripts namespace, but for this script we pass a parameter called `resource_namespace` at run time to indicate the namespace in which the pods we need to exec to. So unless there's some magic that I don't yet understand (can we somehow provide it for `resource_namespace` dynamically?), we need to give this permission at the cluster scope so that it will have that permission in the `resource_namespace` namespace. WDYT?

**Pull request merged**
